pkg/controller/common.go

```diff
@@ -1529,29 +1527,44 @@ func (c *SharedState) createReplicas(ctx context.Context, remainingReplicas int,
 		remainingReplicas--
 	}
 
+	for _, podObj := range podObjs {
+		azr.getSharedState().eventRecorder.Eventf(podObj.DeepCopyObject(), v1.EventTypeNormal, ReplicaAttachmentCreationSuccessEvent, "%d replica attachment(s) created for volume %s", requiredReplicas-remainingReplicas, volumeName)
+	}
+
 	if remainingReplicas > 0 {
 		//no failed replica attachments, but there are still more replicas to reach MaxShares
 		request := ReplicaRequest{VolumeName: volumeName, Priority: remainingReplicas}
 		c.priorityReplicaRequestsQueue.Push(&request)
+		for _, podObj := range podObjs {
+			azr.getSharedState().eventRecorder.Eventf(podObj.DeepCopyObject(), v1.EventTypeWarning, ReplicaAttachmentCreationFailedEvent, "%d/%d replica attachment(s) for volume %s could not be created. Not enough suitable nodes", remainingReplicas, requiredReplicas, volumeName)
```

Comment:
Consider "There are not enough suitable nodes for %d of the %d requested replica mounts for volume %s."

Reply:
Updated the event message.